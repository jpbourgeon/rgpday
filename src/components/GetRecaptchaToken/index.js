import React from 'react'
import { ReCaptcha } from 'recaptcha-v3-react'

class Contact extends React.Component {
  constructor (props) {
    super(props)
    this.state = { token: '' }
    this.recaptcha = React.createRef()
    this.getNewRecaptcha = this.getNewRecaptcha.bind(this)
    this.handleReCaptchaToken = this.handleReCaptchaToken.bind(this)
  }

  getNewRecaptcha () {
    this.recaptcha.current.execute()
  }

  handleReCaptchaToken (token) {
    this.setState({ token })
  }

  render () {
    return (
      <code style={{ padding: '1em' }}>
        <h1>Get ReCaptcha tokens</h1>
        <div><button onClick={this.getNewRecaptcha}>Reload</button>
          <ReCaptcha
            action='contact'
            sitekey='6LedLpMUAAAAAG8Ai4M4x9wTcIs4rPmvYV82a7Yh'
            verifyCallback={this.handleReCaptchaToken}
            ref={this.recaptcha}
          />
        </div><br />
        <div>{this.state.token}</div>
      </code>
    )
  }
}

export default Contact
