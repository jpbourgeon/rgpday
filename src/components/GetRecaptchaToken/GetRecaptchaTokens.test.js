/* global describe, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import GetRecaptchaToken from './'
import jssSerializer from '../../utils/jssSnapshotSerializer'

expect.addSnapshotSerializer(jssSerializer)

describe('The GetRecaptchaToken component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<GetRecaptchaToken />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
