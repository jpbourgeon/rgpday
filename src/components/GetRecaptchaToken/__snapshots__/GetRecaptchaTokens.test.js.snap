// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The GetRecaptchaToken component should render correctly 1`] = `
<code
  style={
    Object {
      "padding": "1em",
    }
  }
>
  <h1>
    Get ReCaptcha tokens
  </h1>
  <div>
    <button
      onClick={[Function]}
    >
      Reload
    </button>
  </div>
  <br />
  <div>
    
  </div>
</code>
`;
